### DISCUSSION TITLE
Ever wanted to use e107 admin on a small screen?

### DISCUSSION THREAD

**Discussion by Alex-e107nl** [e107inc/e107#5165] [General]
You can now... look at [https://github.com/Alex-e107nl/bootstrap3-mobile-admin](https://github.com/Alex-e107nl/bootstrap3-mobile-admin) 

Have you ever tried to manage e107 on your ipad mini of phone? Now you can, have a simple and fast responsive menu with everything accessible within 3 mouse clicks :-) 1 core template with changes, unfortunately that is not possible within the theme. (admin_template.php)
![e107_mobile_admin](https://github.com/e107inc/e107/assets/11962513/2ff8e4c8-d757-49b6-98df-e676f4afc710)

Of course it is not completely finished yet, but it is already very usable, help is always nice!

Wishing everyone a good and healthy 2024!

**Comment by Moc** [comment]
Nice work!

You're very welcome to submit pull requests to make changes to core. This way it doesn't need to be a separate repository and everyone can benefit.

**Reply by Alex-e107nl** [reply]
Thank you Moc, it is free, anyone can download it, you should really see this as a concept, it already works fine but a lot still needs to be done, in my opinion there is a lot of duplicates in the css files and I would like a solution for customizing the admin template, why can't I just copy it like the other core templates and place it in the theme folder?

**Reply by Jimmi08** [reply]
@Alex-e107nl you wrote this before me :)

@Moc, if you can't put admin_template.php in bootstrap3 theme but it uses e107::getTemplate(), it is bug for me. This should be fixed at first.
This is a nice attempt, but it only works with the dark version and unfortunately, it didn't work for me fully as I supposed ( maybe the wrong admin skin and admin layout).  There is the option to display labels with admin navigation too...  
And @Alex-e107nl big thanks, I have been complaining about this for years :)

**Reply by Moc** [reply]
@Alex-e107nl I meant that your changes can be incorporated into core so they're available in the next release of e107, rather than someone having to download them separately afterwards. I haven't done any testing personally, but the admin area should already be responsive for a variety of screen sizes. If not, then it should be improved in core. 


As for customising the admin theme. The decision was made to not allow any customisation on the admin area to avoid any unintended changes that break functionality, making it rather hard to fix for a regular user. Please submit any feature requests or proposed changes as new issues, so we can improve the admin area/theme further. 

Unfortunately admin_template is poorly named and place in the file structure. It is not a template file that is intended to be edited. This is something that should be made clearer.

**Reply by Alex-e107nl** [reply]
Moc, a relation of mine always works on his iPad to update his e107 website and he indicated that it was almost impossible, that is why I adjusted it, you can easily test everything yourself and then see that up to 2.3.1 e107 was indeed responsive and worked well on a small screen, the admin_template.php was almost empty then!

If you now look at your own e107 website and reduce the screen size, you will see exactly what the problem is. There is probably a dev who can perhaps solve this very quickly in an intelligent way, and not being allowed to adjust the admin screen is not a problem in itself, but then there must also be someone who deals with that and ensures that when BS3 is disappearing, the admin has been converted to BS5 and everything is correct, just like with the tinymce version, 4 is really too old and is no longer supported...

But I had also asked you to take a look at what I tried to add to e107 and why it did not work, the plupload problem is still in e107 while it is really a very simple change, but anyone who works with e107 in another language has an error message in the code on the admin side that is not necessary.

**Reply by Jimmi08** [reply]
@Alex-e107nl I am afraid that some admin scripts depend on Bootstrap 3 javascript.

**Reply by Moc** [reply]
@Alex-e107nl I just tested with a clean install (latest Github) on Chrome and the Admin Area seems to be responsive properly. What's not working for you? Please submit a new bug report with screenshots so this can be fixed this in core.

**Reply by Alex-e107nl** [reply]
@Moc Which admin theme do you use Tijn? Modern Dark - A high-contrast dark skin?
Github of today, is this what you mean with responsive properly? (php 8.1) Maybe you see something else?
![e107_smallscreen_2024-04-11](https://github.com/e107inc/e107/assets/11962513/9837f590-2045-423a-a43e-1fa1df02fcc8)

**Reply by Moc** [reply]
@Alex-e107nl I see it now. Definitely needs re-working. Please post it up as a bug and reference this discussion.

**Comment by CaMer0n** [comment]
@Alex-e107nl I love the work you did there! It looks much better than what is currently in the core. 
Question: Does the desktop layout still function normally with the changes you made?

**Reply by Alex-e107nl** [reply]
Hello Cameron, thank you, the admin theme works normally when you use it on a desktop, when you use it for the first time just clear the buffers (ctrl f5) and then you will see no difference on a large screen except for the home button that is now a regular dropdown.

I use it in all my e107 websites and it works perfectly!

![adminfullscreen_e107](https://github.com/e107inc/e107/assets/11962513/5402ea3a-7049-4cf1-9f72-9c4b63378cc6)

**Comment by CaMer0n** [comment]
Great stuff! I found a couple of minor issues on desktop. I'll add them to your repo issue tracker.
